src/DssSpell.t.base.sol

```diff
@@ -221,11 +213,11 @@ contract DssSpellTestBase is Config, Test {
     }
 
     function _concat(string memory a, string memory b) internal pure returns (string memory) {
-        return string(abi.encodePacked(a, b));
+        return string.concat(a, b);
     }
 
     function _concat(string memory a, bytes32 b) internal pure returns (string memory) {
-        return string(abi.encodePacked(a, _bytes32ToStr(b)));
+        return string.concat(a, _bytes32ToStr(b));
```

Comment:
Just going to dig in here, not raising an issue directly

So abi.encodePacked will pack the values (remove leading zeroes) whereas I don't think string.concat does

See https://docs.soliditylang.org/en/v0.8.17/units-and-global-variables.html?highlight=string%20concat#members-of-string

And https://docs.soliditylang.org/en/v0.8.17/units-and-global-variables.html?highlight=abi.encodepacked#abi-encoding-and-decoding-functions
<img width="587" alt="Image" src="https://user-images.githubusercontent.com/103920908/212319462-31666813-34a2-4769-9f32-36a6f610fe8b.png">

I don't think there are going to be any issues here since string.concat gets given two strings (we cast using `string()` in `_bytes32ToStr`) but I wanted to confirm that this has been considered and the outputs are never different (I don't think they ever are).

Reply:
good catch. It seems to me they would be the same since as you say we're giving it 2 strings.  Can you come up with a test that would prove this?

Reply:
I agree that it would be the same.

We *could* in theory test this by generating random bytes strings (with leading zeroes) that get cast to string and then comparing the output, but (1) generating these strings would be very painful (even in Foundry) and (2) it doesn't feel like it fits within scope of what is contained in this repo.

I think if we want to do a side investigation on this, that would be a good idea, but it is probably not suited here. 

Again I *am basically certain* that our outputs should be the same but it's good to consider edge cases.

Reply:
https://github.com/The-Arbiter/to_string_or_not_to_string